nemo/collections/llm/recipes/optim/adam.py

```diff
@@ -1,16 +1,34 @@
+import torch
 from megatron.core.optimizer import OptimizerConfig
 
-from nemo import lightning as nl
-from nemo.collections.llm.utils import factory
+from nemo.collections.llm.utils import Config
+from nemo.lightning.pytorch.optim import CosineAnnealingScheduler, MegatronOptimizerModule, OptimizerModule
 
 
-@factory
-def adam_with_cosine_annealing() -> nl.OptimizerModule:
-    return nl.MegatronOptimizerModule(
-        config=OptimizerConfig(optimizer="adam", lr=0.001, use_distributed_optimizer=True),
-        lr_scheduler=nl.lr_scheduler.CosineAnnealingScheduler(),
+def distributed_fused_adam_with_cosine_annealing() -> Config[OptimizerModule]:
```

Comment:
some of these don't match https://gitlab-master.nvidia.com/dl/JoC/nemo-ci/-/issues/513

Reply:
Updated in https://github.com/NVIDIA/NeMo/pull/10147/commits/d6fd76076addb91fefb356b1de0494539c65212a